### ISSUE TITLE
Codist incompatible with Supercharger

### ISSUE THREAD

**Issue by IneedHelp** [wmjordan/Codist#31]
Hello, I absolutely love your extension and I use it for the enhanced tooltips (Super Quick Info), but after installing it, I have realized that the Supercharger "magic comments" aren't working anymore and those are extremely important for me as well.

The only way to make Supercharger's magic comments work again is to completely disable the Codist extension, disabling its features on the General page doesn't have any effect.

I'll have to prioritize Supercharger because losing all the comment formatting dependent on it is much worse than not having enhanced QuickInfo tooltips. Hope this can be solved, thank you!

Supercharger links:
http://supercharger.tools/#sectionMagicComments
https://marketplace.visualstudio.com/items?itemName=MichaelKissBG8.Supercharger

**Comment by wmjordan** [comment]
Hi,

Thank you for posting the issue.
I once tried SuperCharger for a little time and tested its compatibility with Codist.
I did not notice that the magic comments were incompatible, since I prioritized my own syntax highlighter over its. Please try to disable the **Syntax Highlight** feature of Codist in the _General_ options page of Codist and **reopen the document windows**. This should solve your problem.

**Comment by IneedHelp** [comment]
Hello,

In Codist, with only Super Quick Info enabled, Supercharger's magic comments doesn't work, even if reopening the documents or reloading the project/solution. As for Supercharger, it just has a toggle button for enabling/disabling magic comments, but I've tried with magic comments disabled several times and Codist's syntax highlight on and it still didn't work. The only way to make it work is to one one or the other completely disabled as VS extensions.

**Comment by wmjordan** [comment]
I reinstalled SuperCharger.
The Magic Comments could not work, really.

Could you also hit the Technical Support command in the SuperCharger menu and find their email address to report the issue to them?

**Comment by IneedHelp** [comment]
I have tried to contact them already, but do you think it would be possible for you to make your extension completely shut down all components that are not selected so that the Codist syntax highlighting doesn't conflict with other syntax highlighting while the Super Quick Info still remains active?

**Comment by IneedHelp** [comment]
Would it be possible for you to entirely strip the syntax highlighting and compile a test version without it that I could test?

**Comment by wmjordan** [comment]
After some investigations, I narrowed down the source of conflicts to the code comment classification definitions, which could not be toggled at runtime, therefore you could not make Magic Comment work unless you disable the whole Codist.

Although Codist and SuperCharger do not share a classification definitions with a same name, the SuperCharger's magic comment just does not work. I could not figure out what was happening within SuperCharger.

Before SuperCharger's author replies and makes necessary modifications, you may use this version which has disabled the code comment classification definitions. Note: DO NOT click the Comment part of Codist's syntax highlight option page, otherwise VS will crash.
[Codist.zip](https://github.com/wmjordan/Codist/files/2323934/Codist.zip)

**Comment by IneedHelp** [comment]
Excellent, I've just installed the version you provided and it works great. No longer conflicting with the Supercharger comments.

I did email the Supercharger support and pointed to this issue, if I get any response, I will surely reply back.

For the meantime, thank you very much, I will use this version you've provided. Also, the reason I have discovered this extension is that after I have updated ReSharper, the Enhanced Tooltip extension ( https://github.com/MrJul/ReSharper.EnhancedTooltip ) wasn't working anymore as it needs updated, so I looked for alternatives and found Codist. Now I absolutely love the Super Quick Info and I don;t think I'll go back to Enhanced Tooltip even if it does get updated. I will open a couple threads with some feature requests I think would slightly improve the feel of Super Quick Info. Thank you!

**Comment by wmjordan** [comment]
Would you mind telling me what kind of annotations of Magic Comments you are using?

**Comment by wmjordan** [comment]
That version posted above would never be considered as a published release, since it had disabled the comment tagger completely. We have to find out why SuperCharger refuses to work when Codist exported comment related definitions.

**Comment by IneedHelp** [comment]
Yeah, I figured you wouldn't make that version a public release since its comment tagger is hacked off. I'm still waiting to see if the people over Supercharger support say anything and if not, I'll try to press the issue further.

**Comment by wmjordan** [comment]
Please comment here if they reply or provide a new version which no longer conflicts with Codist.
